contracts/bonding/BondingManager.sol

```diff
@@ -186,14 +176,12 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         if (transcoderStatus(msg.sender) == TranscoderStatus.NotRegistered) {
             if (transcoderPool.isFull()) {
                 address lastTranscoder = transcoderPool.getLast();
+                // Must have more stake than the last transcoder in the pool
+                require(delegatedAmount > transcoderPool.getKey(lastTranscoder));
```

Comment:
Previously, if the caller did not have enough delegated stake to join the transcoder pool (the transcoder pool is full and the caller's delegated stake <= the delegated stake of the transcoder at the bottom of the pool), this function would just fail silently (fields are updated, but the transcoder would not actually be registered in the transcoder pool). Now, this function reverts if the caller does not have enough delegated stake to join the transcoder pool

Reply:
I *thought* that it was ok if this transaction failed silently because the caller would still be registered candidate transcoder so they would be eligible to receive other stake and eventually make it into the pool, even if their initial transaction didn't have enough stake to make it into the pool.

If we revert here, then it's impossible to get others to delegate towards you (sponsor you) into the pool since you need enough stake in possession to make it on your own.

Or was I misunderstanding how it worked before?

Reply:
The function actually uses the caller's delegated amount (which includes the amount other people have delegated to the caller) as opposed to its bonded amount (which is just the amount that the caller has bonded by itself) so it would still be possible for other people to "sponsor" you so you could have enough delegated stake to join the pool. Once you have enough people sponsoring you such that you have a sufficient delegated stake to join the pool, you can call `transcoder` to evict the transcoder with the lowest delegated stake in the pool and join the pool.

Before this change, the transaction would fail silently such that you end up not being in the transcoder pool (and thus not deemed a registered transcoder because `transcoderStatus`  considers you registered only if you are in the transcoder pool), but you still updated your reward cut, fee share, price, etc.

Reply:
Agreed. @yondonfu created an issue #191 where we can maybe try and figure out what that could mean as far as implementation goes.